asapdiscovery-ml/asapdiscovery/ml/cli.py

```diff
@@ -788,6 +789,258 @@ def build_e3nn(
+    cont: bool | None = None,
+    loss_dict: dict | None = None,
+    device: torch.device | None = None,
+    overwrite_trainer_config_cache: bool = False,
+    overwrite_optimizer_config_cache: bool = False,
+    overwrite_model_config_cache: bool = False,
+    overwrite_es_config_cache: bool = False,
+    overwrite_ds_config_cache: bool = False,
+    overwrite_ds_cache: bool = False,
+    overwrite_ds_split_config_cache: bool = False,
+    overwrite_loss_config_cache: bool = False,
+):
+    # First check if Trainer cache exists and skip everything else if so
+    if (
+        trainer_config_cache
+        and trainer_config_cache.exists()
+        and (not overwrite_trainer_config_cache)
+    ):
+        print("loaded trainer from cache", flush=True)
+        t = Trainer(**json.loads(trainer_config_cache.read_text()))
+    else:
+        # Build each dict and pass to Trainer
+        optim_config = {
+            "cache": optimizer_config_cache,
+            "overwrite_cache": overwrite_optimizer_config_cache,
+            "optimizer_type": optimizer_type,
+            "lr": lr,
+            "weight_decay": weight_decay,
+            "momentum": momentum,
+            "dampening": dampening,
+            "b1": b1,
+            "b2": b2,
+            "eps": eps,
+            "rho": rho,
+        }
+        model_config = {
+            "cache": model_config_cache,
+            "overwrite_cache": overwrite_model_config_cache,
+            "model_type": ModelType.visnet,
+            "rand_seed": model_rand_seed,
+            "grouped": grouped,
+            "strategy": strategy,
+            "pred_readout": pred_readout,
+            "combination": combination,
+            "comb_readout": comb_readout,
+            "max_comb_neg": max_comb_neg,
+            "max_comb_scale": max_comb_scale,
+            "pred_substrate": pred_substrate,
+            "pred_km": pred_km,
+            "comb_substrate": comb_substrate,
+            "comb_km": comb_km,
+            "lmax": lmax,
+            "vecnorm_type": vecnorm_type,
+            "trainable_vecnorm": trainable_vecnorm,
+            "num_heads": num_heads,
+            "num_layers": num_layers,
+            "hidden_channels": hidden_channels,
+            "num_rbf": num_rbf,
+            "trainable_rbf": trainable_rbf,
+            "max_z": max_z,
+            "cutoff": cutoff,
+            "max_num_neighbors": max_num_neighbors,
+            "vertex": vertex,
+            "reduce_op": reduce_op,
+            "mean": mean,
+            "std": std,
+            "derivative": derivative,
+        }
+        if (es_config_cache and es_config_cache.exists()) or es_type:
+            es_config = {
+                "cache": es_config_cache,
+                "overwrite_cache": overwrite_es_config_cache,
+                "es_type": es_type,
+                "es_patience": es_patience,
+                "es_n_check": es_n_check,
+                "es_divergence": es_divergence,
```

Comment:
adjust these as in the methods for the other models (also see my note in `config.py`)